lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -651,6 +651,24 @@ function dosomething_global_url($path, $options = NULL) {
 }
 
 /**
+ * Build a global node URL that adds the prefix and a fragment
+ *
+ * @param string $nid
+ *    The nid of the node
+ * @param string $language
+ *    The language of the URL
+ * @param string $fragment
+ *    Optional fragment to append
+ */
+function dosomething_global_node_url($nid, $language, $fragment = NULL) {
+  $raw_url = 'node/' . $nid;
+  if (isset($fragment)) {
+    $raw_url .= '#' . $fragment;
```

Comment:
I'm doing this because the url function has been the cause of every problem we've been seeing regarding bad URL's, it doesn't play well with languages and prefixes

this is the method I've been using for quite a while now in a few places and it's been working great